ui/src/main/webapp/lib/components/AttributeMapper.js

```diff
@@ -0,0 +1,130 @@
+import { Card, CardHeader } from 'material-ui/Card'
+import AutoComplete from 'material-ui/AutoComplete'
+
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn
+} from 'material-ui/Table'
+
+import Title from 'components/Title'
+import Description from 'components/Description'
+
+class AttributeMapper extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      subjectClaim: '',
+      userAttribute: '',
+      selected: []
+    }
+  }
+  handleRemove () {
+    const { onChange, attributeMappings = [] } = this.props
+
+    onChange(attributeMappings
+      .filter((_, i) => !this.state.selected.includes(i)))
+
+    this.setState({ selected: [] })
+  }
+  handleAdd () {
+    const { onChange, attributeMappings = [] } = this.props
+    const newPair = {
+      key: this.state.subjectClaim,
+      value: this.state.userAttribute
+    }
+
+    onChange(attributeMappings.filter(({ key }) => key !== newPair.key)
+      .concat(newPair))
+
+    this.setState({ subjectClaim: '', userAttribute: '' })
+  }
+  render () {
+    const {
+      title,
+      description,
+      attributeMappings = [],
+      subjectClaims = [],
+      userAttributes = []
+    } = this.props
+
+    return (
+      <div>
+        <SelectField
+          fullWidth
+          value={this.state.subjectClaim}
+          onChange={(e, i, subjectClaim) => this.setState({ subjectClaim })}
+          floatingLabelText='STS Claim'>
+          {subjectClaims.map((claim, i) =>
+            <MenuItem key={i} value={claim} primaryText={claim} />)}
+        </SelectField>
+
+        <AutoComplete
+          menuStyle={{maxHeight: 200, overflowY: 'scroll'}}
+          fullWidth
+          openOnFocus
+          dataSource={userAttributes.map((value) => ({text: String(value), value}))}
+          filter={AutoComplete.noFilter}
+          floatingLabelText='User Attribute'
+          searchText={this.state.userAttribute}
+          onNewRequest={({ value }) => { this.setState({ userAttribute: value }) }}
+          onUpdateInput={(userAttribute) => { this.setState({ userAttribute }) }} />
+
+        <RaisedButton
+          primary
+          fullWidth
+          style={{ margin: '20px 0' }}
+          label='Add Mapping'
+          onClick={this.handleAdd.bind(this)}
+          disabled={this.state.subjectClaim === '' || this.state.userAttribute === ''} />
+
+        <Card expanded style={{ width: '100%' }}>
+          <CardHeader style={{ fontSize: '0.80em' }}>
+            <Title>{title}</Title>
+            <Description>{description}</Description>
+          </CardHeader>
+          <Table multiSelectable onRowSelection={(selected) => this.setState({ selected })}>
+            <TableHeader displaySelectAll={false}>
+              <TableRow>
+                <TableHeaderColumn>STS Claim</TableHeaderColumn>
+                <TableHeaderColumn style={{ width: 120 }}>User Attribute</TableHeaderColumn>
```

Comment:
It looks like this is being used by the LDAP wizard as well. We may want to be more general with our variable names and make the labels props.